src/Cake.SqlServer/SqlDacpacImpl.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+using Microsoft.SqlServer.Dac;
+
+namespace Cake.SqlServer
+{
+    internal static class SqlDacpacImpl
+    {
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="settings">The settings.</param>
+        public static void ExtractDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, ExtractDacpacSettings settings)
+        {
+            context.Log.Information($"About to extract a dacpac file from database {targetDatabaseName}");
+
+            var service = new DacServices(connectionString);
+
+            service.Extract(settings.OutputFile.FullPath, targetDatabaseName, settings.Name, settings.Version, settings.Description, settings.Tables);
+
+            context.Log.Information($"Finished creating dacpac file from database {targetDatabaseName}. File location is {settings.OutputFile}");
+        }
+
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="dacpacFilePath">Path to the dac file.</param>
+        /// <param name="config">Configure the sql deployment</param>
+        public static void PublishDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, string dacpacFilePath, Action<PublishOptions> config = null)
```

Comment:
I would much rather not to pass `Action<>`. And also I'd rather not force users to take dependency on `Microsoft.SqlServer.Dac` namespace - they will have to add this namespace when they create `PublishOptions` or `Action<PublishOptions>`.  How problematic it would to have a POCO settings class living in our namespace that represents all the properties you need for dacpac publish? Similar to what you do with `ExtractDacpacSettings`?